### DISCUSSION TITLE
Current Index of Spotlight

### DISCUSSION THREAD

**Discussion by kylelol** [bumble-tech/appyx#603] [Q&A]
Is it possible to access the current `activeIndex` of `Spotlight`? My use case is I have a `ParentNode` that uses a `Scaffold` with a `bottomBar` and I want to update the `selected` parameter of the `NavigationBarItem`.

My other idea was to just track the change in `resolve`.

**Comment by KovalevAndrey** [comment]
Hi @kylelol. If you're using 1.x version you could use `com.bumble.appyx.navmodel.spotlight.activeIndex()` extension method on your `spotlight`. It will return a flow of currently selected index which you can observe

**Reply by kylelol** [reply]
Is that also available in 2.x?

**Reply by KovalevAndrey** [reply]
@kylelol it's not a part of the library yet but you can observe model yourself:

```
private val model: SpotlightModel<InteractionTarget> = SpotlightModel( ... ),

private val spotlight: Spotlight<InteractionTarget> = Spotlight(
        model = model,
        ...
)


// in compose
val currentIndex by model
    .output
    .mapState(rememberCoroutineScope()) { it.currentTargetState.activeIndex }
    .collectAsState()

```

**Reply by RFonzi** [reply]
Our team appreciated the simplicity of `spotlight.activeIndex()`/`spotlight.current()` from 1.x. We managed to find the equivalent here in 2.x via `output` on our own, but it was much less discoverable IMO. But I'm aware this is still an alpha 😅
